clinica/utils/dwi.py

```diff
@@ -1,253 +1,351 @@
 
-    Args:
-        in_file (str): The b0 volumes already registered.
-        out_file (str, optional): Name of the file. Defaults to None.
+    in_bval : PathLike, optional
+        The path to the bval file. This will be used to filter the volumes.
+        Only volumes for which bval is less than low_bval will be kept for computations.
+        If None, all volumes are kept. Default is None.
 
-    Returns:
-        The mean of the b0 volumes.
+    low_bval : float, optional
+        The threshold to apply to bvalues to get the volumes which should be kept.
+        Default=5.0.
+
+    squeeze : bool, optional
+        Whether the returned volume should be squeezed or not.
+        If True, it will be a 3D array, if False, a 4D array with a dummy 4th dimension.
+        Default=False
 
-    Warnings:
-        The b0 volumes must be registered.
+    out_file : str, optional
+        Name of the output file.
+        If None, the output file will be built from the input DWI file base with
+        the suffix '_avg_b0.nii[.gz]'. Defaults to None.
+
+    Returns
+    -------
+    out_file : str
+        The path to the nifti image file containing the mean of the b0 volumes.
+
+    Raises
+    ------
+    FileNotFoundError:
+        If the DWI input file does not exist.
+
+    ValueError:
+        If low_bval < 0.
     """
-    import os
+    from pathlib import Path
 
-    import nibabel as nb
-    import numpy as np
+    from clinica.utils.image import compute_aggregated_volume, get_new_image_like
 
-    if not out_file:
-        fname, ext = os.path.splitext(os.path.basename(in_file))
-        if ext == ".gz":
-            fname, ext2 = os.path.splitext(fname)
-            ext = ext2 + ext
-        out_file = os.path.abspath(f"{fname}_avg_b0{ext}")
+    in_dwi = Path(in_dwi)
 
-    imgs = np.array(nb.four_to_three(nb.load(in_file)))
-    b0s = [im.get_fdata(dtype="float32").astype(np.float32) for im in imgs]
-    b0 = np.average(np.array(b0s), axis=0)
+    if not in_dwi.exists():
+        raise FileNotFoundError(f"DWI file not found : {in_dwi}.")
 
-    hdr = imgs[0].get_header().copy()
-    hdr.set_data_shape(b0.shape)
-    hdr.set_xyzt_units("mm")
-    hdr.set_data_dtype(np.float32)
-    nb.Nifti1Image(b0, imgs[0].get_affine(), hdr).to_filename(out_file)
+    if low_bval < 0:
+        raise ValueError(f"low_bval should be >=0. You provided : {low_bval}.")
+
+    out_file = out_file or add_suffix_to_filename(in_dwi, "avg_b0")
+    volume_filter = get_b0_filter(in_bval, low_bval=low_bval) if in_bval else None
+    b0 = compute_aggregated_volume(in_dwi, np.average, volume_filter)
+    if not squeeze:
+        b0 = b0[..., np.newaxis]
+    b0_img = get_new_image_like(in_dwi, b0)
+    b0_img.to_filename(out_file)
 
     return out_file
 
 
-def b0_dwi_split(in_dwi: str, in_bval: str, in_bvec: str, low_bval: float = 5.0):
-    """Split DWI dataset.
+def check_dwi_dataset(dwi_dataset: DWIDataset) -> DWIDataset:
```

Comment:
I improved the docstring and added Parameters and returns sections.
For very simple functions like this one (or the ones just after), where the inputs and outputs are straightforward, it can be less verbose to omit them.

Reply:
Alright !